File: 24.04 Test tasks/task3.py
import math

value = int(input())

mod = 0

sum = 0

arr = []

while(value!=0):
    mod = value % 3
    arr.append(mod)
    arr.append(mod)
    value = math.floor(value / 3)

for i in range(len(arr)):
    sum += (arr[i] * pow(3,i))

print(sum)

File: 24.04 Test tasks/task3.cpp


#include <iostream>
#include <list>
#include <cmath>
using namespace std;

int main()
{
	int value, mod = 0, sum = 0, count = 0;
	list<int> code;
	cin >> value;

	while (value != 0) {
		mod = value % 3;
		code.push_back(mod);
		code.push_back(mod);
		value = floor(value / 3);


	};
	
	for (auto i = code.begin(); i!=code.end(); ++i)
	{
		sum += (*i * pow(3, count));
		count++;
	}

	cout << sum << endl;
}

